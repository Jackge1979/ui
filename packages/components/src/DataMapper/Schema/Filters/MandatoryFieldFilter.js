import Filter from './Filter';

export const ID = 'mandatory-field-filter';

export default class MandatoryFieldFilter extends Filter {

  constructor(active) {
    super(ID, active);
  }

  select(dataAccessor, schema, element) {
    if (this.active) {
      return dataAccessor.isElementMandatory(element);
    }
    return true;
  }

}
