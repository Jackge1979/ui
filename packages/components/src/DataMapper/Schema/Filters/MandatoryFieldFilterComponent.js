import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Toggle from '../../../Toggle';

/**
* Component used for the MandatoryFieldFilter filter.
* It displays a toggle button with a label.
*/
export default class MandatoryFieldFilterComponent extends Component {
	constructor(props) {
		super(props);
		this.onChange = this.onChange.bind(this);
	}

	onChange() {
		const active = this.props.filter.isActive();
		this.props.filter.setActive(!active);
		this.props.onFilterChange(this.props.filter);
	}

	render() {
		const { filter } = this.props;
		return (
			<Toggle
				id="show-mandatory-fields"
				onChange={this.onChange}
				label="Show Mandatory Fields (*) Only"
				checked={filter.isActive()}
			/>
		);
	}
}

MandatoryFieldFilterComponent.propTypes = {
	filter: PropTypes.object,
	onFilterChange: PropTypes.func,
};
