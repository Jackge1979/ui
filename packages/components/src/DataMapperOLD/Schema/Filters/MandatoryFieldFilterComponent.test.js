import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import MandatoryFieldFilterComponent from './MandatoryFieldFilterComponent';
import MandatoryFieldFilter from './MandatoryFieldFilter';

it('mandatory-field-filter-component', () => {
	const filter = new MandatoryFieldFilter(false);
	// create React tree
	const tree = renderer.create(<MandatoryFieldFilterComponent filter={filter} />).toJSON();
	expect(tree).toMatchSnapshot();
});

it('mandatory-field-filter-component activate', () => {
	const filter = new MandatoryFieldFilter(false);
	const onFilterChange = jest.fn();

	const filterComp = (
		<MandatoryFieldFilterComponent filter={filter} onFilterChange={onFilterChange} />
	);
	const wrapper = mount(filterComp);
	wrapper.find('input').simulate('change');

	expect(onFilterChange).toBeCalled();
});
