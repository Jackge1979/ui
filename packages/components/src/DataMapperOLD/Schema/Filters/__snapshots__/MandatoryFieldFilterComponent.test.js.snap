// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mandatory-field-filter-component 1`] = `
<div
  className="checkbox tc-toggle switch checkbox"
>
  <label
    data-feature={undefined}
    htmlFor="show-mandatory-fields"
  >
    <input
      autoFocus={undefined}
      checked={false}
      disabled={false}
      id="show-mandatory-fields"
      onBlur={undefined}
      onChange={[Function]}
      type="checkbox"
    />
    <span>
      Show Mandatory Fields (*) Only
    </span>
  </label>
</div>
`;
